images/base/scripts/common/setup.sh

```diff
@@ -83,7 +83,12 @@ fi
 
 # First install protoc
 cd /tmp
-git clone https://github.com/google/protobuf.git
+if [ x$MACHINE = xs390x ]
+then
+    git clone https://github.com/gongsu832/protobuf.git
```

Comment:
The same applies for patching yourself. If you checkout a different branch, the patch may not work and may need to be updated. So yeah, you need to coordinate with the patch maintainer either way if you are going to change the branch you checkout.